devices/ePaper/Buffers/FrameBuffer2BitPerPixel.cs

```diff
@@ -0,0 +1,245 @@
+            set => this.BlackBuffer.CurrentFramePage = this.ColorBuffer.CurrentFramePage = value;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FrameBuffer2BitPerPixel"/> class.
+        /// </summary>
+        /// <param name="height">The height of the frame to manage.</param>
+        /// <param name="width">The width of the frame to manage.</param>
+        public FrameBuffer2BitPerPixel(int height, int width)
+        {
+            this.Height = height;
+            this.Width = width;
+
+            this.BlackBuffer = new FrameBuffer1BitPerPixel(height, width);
+            this.ColorBuffer = new FrameBuffer1BitPerPixel(height, width);
+        }
+
+        /// <inheritdoc/>>
+        public void Clear()
+        {
+            this.BlackBuffer.Clear(Color.White);
+            this.ColorBuffer.Clear(Color.Black); // black sets it to 0s
+        }
+
+        /// <inheritdoc/>>
+        public void Clear(Color color)
+        {
+            this.BlackBuffer.Clear(color);
+            this.ColorBuffer.Clear(color);
+        }
+
+        /// <inheritdoc/>>
+        public void Fill(Color color)
+            => this.Fill(Point.Zero, this.Width, this.Height, color);
+
+        /// <inheritdoc/>
+        public void Fill(Point start, int width, int height, Color color)
+        {
+            if (!this.IsPointWithinFrameBuffer(start))
+                return;
+
+            // if the current color isn't black or white, then we send data to the color buffer
+            if (color != Color.Black
+                && color != Color.White)
+            {
+                this.ColorBuffer.Fill(start, width, height, Color.White); // white will write 1 to the buffer
+                this.BlackBuffer.Fill(start, width, height, Color.Black); // black writes 0s to the buffer
+            }
+            else if (color == Color.Black)
+            {
+                this.BlackBuffer.Fill(start, width, height, Color.Black);
+                this.ColorBuffer.Fill(start, width, height, Color.Black); // black will write 0 to the buffer
+            }
+            else
+            {
+                // white
+                this.BlackBuffer.Fill(start, width, height, Color.White);
+                this.ColorBuffer.Fill(start, width, height, Color.Black); // black will write 0 to the buffer
+            }
+        }
+
+        /// <inheritdoc/>
+        /// <exception cref="InvalidOperationException" />
+        /// <remarks>
+        /// This method is not implemented for this class as it is not possible 
+        /// to convert a byte from <see cref="BlackBuffer"/> and <see cref="ColorBuffer"/>
+        /// into a <see cref="Color"/> struct. We don't know what color <see cref="ColorBuffer"/>
+        /// represents in the actual display hardware.
+        /// </remarks>
+        public Color GetPixel(Point point)
+        {
+            throw new InvalidOperationException();
```

Comment:
while I do understand the description, why not still converting with 2 different colors taken arbitrararly?